### ISSUE TITLE
Do not export private fields

### ISSUE THREAD

**Issue by RazrFalcon** [eqrion/cbindgen#149]
```rust
#[repr(C)]
pub struct my_struct(SomePrivateType);
```
Will be transformed to:
```c
typedef struct {
    SomePrivateType _0;
} my_struct;
```

But should be (I guess):
```c
typedef struct my_struct my_struct;
```

**Comment by RReverser** [comment]
What if there is also a public field? I guess instead of making struct opaque it's better to transform these private fields to byte array or something... but then, C doesn't have public/private modifiers anyway, so this shouldn't matter or help much.

**Comment by eqrion** [comment]
I don't think we should make the struct opaque in this case, as that'd force you to always use it behind a pointer. Which could be unnecessary.

Exporting it as a byte blob with alignment could be an option, but I'm unsure if that's worth the effort.
